File: Data Structures & Algorithms/last-stone-weight/submission-8.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        
        while len(stones) > 1:
            stones.sort()

            x = stones.pop()
            y = stones.pop()

            if x == y:
                stones.append(0)
            if x < y:
                stones.append(y - x)
            if x > y:
                stones.append(x - y)

        return stones[0]

File: Data Structures & Algorithms/validate-parentheses/submission-8.cpp
class Solution {
public:
    bool isValid(string s) {
        unordered_map<char,char> valid = {
            {'}','{'},
            {')','('},
            {']','['}
        };

        stack<char> p_stack;

        for (char c:s) {
            if (c == '}' || c == ']' || c == ')') {
                if (p_stack.empty() || p_stack.top() != valid[c]) {
                    return false;
                } 

                p_stack.pop();
            }

            else if (c == '{' || c == '[' || c == '(') {
                p_stack.push(c);
            }
        }

        if (p_stack.empty()) {
            return true;
        } else {
            return false;
        }
    }
};
